### PR TITLE
Add largest memtable seqno and return it in `OnFlushBegin` event.

### PR DESCRIPTION
**Pull request by 5kbpers**

### CODE REVIEW

File: db/memtable.cc [hunk 1]
```diff
@@ -93,6 +94,7 @@ MemTable::MemTable(const InternalKeyComparator& cmp,
       file_number_(0),
       first_seqno_(0),
       earliest_seqno_(latest_seq),
+      largest_seqno_(0),
```
Comment (tabokie) [inline db/memtable.cc:97]: Shouldn't it be initialized the same as earliest_seqno_?

File: db/memtable.cc [hunk 2]
```diff
@@ -575,6 +580,10 @@ bool MemTable::Add(SequenceNumber s, ValueType type,
         (cur_earliest_seqno == kMaxSequenceNumber || s < cur_earliest_seqno) &&
         !first_seqno_.compare_exchange_weak(cur_earliest_seqno, s)) {
     }
+    uint64_t cur_largest_seqno = largest_seqno_.load(std::memory_order_acquire);
+    while (s > cur_largest_seqno &&
+          largest_seqno_.compare_exchange_weak(cur_largest_seqno, s)) {
```
Comment (tabokie) [inline db/memtable.cc:585]:
```suggestion
          !largest_seqno_.compare_exchange_weak(cur_largest_seqno, s)) {
```

File: db/memtable.cc [hunk 3]
```diff
@@ -10,6 +10,7 @@
 #include "db/memtable.h"
 
 #include <algorithm>
+#include <atomic>
```
Comment (tabokie) [inline db/memtable.cc:13]: Shouldn't need to.

File: db/flush_job.h [hunk 4]
```diff
@@ -38,6 +38,7 @@
 #include "rocksdb/listener.h"
 #include "rocksdb/memtablerep.h"
 #include "rocksdb/transaction_log.h"
+#include "rocksdb/types.h"
```
Comment (tabokie) [inline db/flush_job.h:41]: Shouldn't need it. `SequenceNumber` is already used.

File: db/flush_job.cc [hunk 5]
```diff
@@ -39,6 +39,7 @@
 #include "rocksdb/statistics.h"
 #include "rocksdb/status.h"
 #include "rocksdb/table.h"
+#include "rocksdb/types.h"
```
Comment (tabokie) [inline db/flush_job.cc:42]: No need. Necessary headers are all included in `flush_job.h`.

File: db/db_impl/db_impl_compaction_flush.cc [hunk 6]
```diff
@@ -581,7 +586,7 @@ void DBImpl::NotifyOnFlushBegin(ColumnFamilyData* cfd, FileMetaData* file_meta,
     info.triggered_writes_slowdown = triggered_writes_slowdown;
     info.triggered_writes_stop = triggered_writes_stop;
     info.smallest_seqno = file_meta->fd.smallest_seqno;
-    info.largest_seqno = file_meta->fd.largest_seqno;
+    info.largest_seqno = largest_seqno;
```
Comment (tabokie) [inline db/db_impl/db_impl_compaction_flush.cc:589]: Why `file_meta->fd.largest_seqno` doesn't work?

File: db/db_impl/db_impl.h [hunk 7]
```diff
@@ -51,6 +51,7 @@
 #include "rocksdb/status.h"
 #include "rocksdb/trace_reader_writer.h"
 #include "rocksdb/transaction_log.h"
+#include "rocksdb/types.h"
```
Comment (tabokie) [inline db/db_impl/db_impl.h:54]: Remove.

File: .gitignore [hunk 8]
```diff
@@ -79,3 +79,6 @@ tp2/
 fbcode/
 fbcode
 buckifier/*.pyc
+
+.cache/
+.vscode/
```
Comment (tabokie) [inline .gitignore:84]: Remove these. Better keep it the same as upstream.

File: db/db_impl/db_impl_compaction_flush.cc [hunk 9]
```diff
@@ -18,6 +18,7 @@
 #include "monitoring/perf_context_imp.h"
 #include "monitoring/thread_status_updater.h"
 #include "monitoring/thread_status_util.h"
+#include "rocksdb/types.h"
```
Comment (tabokie) [inline db/db_impl/db_impl_compaction_flush.cc:21]: rm

File: db/db_impl/db_impl_compaction_flush.cc [hunk 10]
```diff
@@ -338,7 +340,8 @@ Status DBImpl::AtomicFlushMemTablesToOutputFiles(
         stats_, &event_logger_, mutable_cf_options.report_bg_io_stats,
         false /* sync_output_directory */, false /* write_manifest */,
         thread_pri));
-    jobs.back()->PickMemTable();
+    SequenceNumber largest_seqno =  jobs.back()->PickMemTable();
```
Comment (Connor1996) [inline db/db_impl/db_impl_compaction_flush.cc:343]: format

**Comment by 5kbpers** [comment]
Close this PR since the main version was switched to 6.29.

File: db/db_impl/db_impl_compaction_flush.cc (hunk 6 above)
Comment (5kbpers) [reply db/db_impl/db_impl_compaction_flush.cc:589]: It was initialized as 0 at the beginning of flush job.

**Pull request closed without merging**